code/modules/clothing/under/accessories/shirts.dm

```diff
@@ -52,4 +52,27 @@
 	name = "t-shirt"
 	desc = "A simple, cheap t-shirt."
 	icon_state = "tshirt"
-	item_state = "tshirt"
\ No newline at end of file
+	item_state = "tshirt"
+
+/obj/item/clothing/accessory/silversun
+	name = "silversun floral shirt"
+	desc = "A stylish Solarian shirt of Silversun design. It bears a floral design. This one is cyan."
+	icon = 'icons/clothing/under/shirts/hawaiian.dmi'
+	icon_state = "hawaii"
+	item_state = "hawaii"
+	contained_sprite = TRUE
+	var/open = FALSE
+
+/obj/item/clothing/accessory/silversun/attack_self(mob/user)
+	open = !open
+	icon_state = "[initial(icon_state)][open ? "_open" : ""]"
+	item_state = icon_state
+	to_chat(user, span("notice", "You [open ? "open" : "close"] \the [src]."))
+	// the below forces the shirt to hard reset its image so it resets later its fucking weird ok
+	inv_overlay = null
+	mob_overlay = null
```

Comment:
Why ain't this a verb? It'd be pretty weird to have to take your shirt off to unbutton it.